mmv1/products/bigquery/api.yaml

```diff
@@ -229,6 +232,35 @@ objects:
               Describes the Cloud KMS encryption key that will be used to protect destination
               BigQuery table. The BigQuery Service Account associated with your project requires
               access to this encryption key.
+      - !ruby/object:Api::Type::String
+        name: type
+        min_version: beta
+        description: |
+          The type of the dataset, one of:
+            DEFAULT - only accessible by owner and authorized accounts,
+            LINKED - linked dataset,
+            EXTERNAL - dataset with definition in external metadata catalog.
+        output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'linkedDatasetSource'
```

Comment:
We'll want to exercise the field in a test in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/third_party/terraform/tests/resource_big_query_dataset_test.go

Reply:
This field can only be used for allowlisted projects at the moment (the request is rejected for not-allowlisted projects). Do the tests always run against the same project? Or is it simply the runner of the tests? If it's the later, I think the test would fail for most users.

Maybe we could make a test that verifies the request is rejected when this field is set? Though we would have to update the test once the field is made publicly usable.

Reply:
Ack- I think we'll want to wait for the feature to become publicly available, in that case.